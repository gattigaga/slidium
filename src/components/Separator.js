import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Line = styled.div`
  width: ${({ orientation }) =>
    orientation === "horizontal" ? "100%" : "0px"};
  height: ${({ orientation }) =>
    orientation === "horizontal" ? "0px" : "100%"};
  margin: ${({ orientation }) =>
    orientation === "horizontal" ? "4px 0px" : "0px 4px"};
  box-sizing: border-box;

  ${({ orientation }) =>
    orientation === "horizontal"
      ? "border-top"
      : "border-right"} : 1px solid #ccc;
`;

const Separator = ({ orientation }) => <Line orientation={orientation} />;

Separator.propTypes = {
  orientation: PropTypes.oneOf(["horizontal", "vertical"])
};

Separator.defaultProps = {
  orientation: "horizontal"
};

export default Separator;
