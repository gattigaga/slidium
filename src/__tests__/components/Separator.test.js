import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Separator from "components/Separator";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Separator />);
});

describe("Icon", () => {
  it("renders in horizontal", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders in vertical", () => {
    wrapper.setProps({
      orientation: "vertical"
    });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
