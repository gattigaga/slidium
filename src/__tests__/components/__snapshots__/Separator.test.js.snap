// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders in horizontal 1`] = `
<styled.div
  orientation="horizontal"
/>
`;

exports[`Icon renders in vertical 1`] = `
<styled.div
  orientation="vertical"
/>
`;
